astropy/table/column.py

```diff
@@ -1120,6 +1124,28 @@ def _copy_attrs_slice(self, out):
         return out
 
     def __setitem__(self, index, value):
+        # Issue warning for string assignment that truncates ``value``
+        if issubclass(self.dtype.type, np.character):
+            # Account for a bug in np.ma.MaskedArray setitem.
+            # https://github.com/numpy/numpy/issues/8624
+            value = np.ma.asanyarray(value, dtype=self.dtype.type)
+
+            # Turn value into a flat masked array with at least 1d
+            vals = np.ma.atleast_1d(value).flatten()
```

Comment:
This will go wrong for >=2-d arrays of strings (which are of course not impossible...)